app/src/main/java/de/tum/in/tumcampusapp/component/tumui/calendar/widget/TimetableWidgetService.kt

```diff
@@ -0,0 +1,118 @@
+package de.tum.`in`.tumcampusapp.component.tumui.calendar.widget
+
+import android.appwidget.AppWidgetManager
+import android.content.Context
+import android.content.Intent
+import android.widget.RemoteViews
+import android.widget.RemoteViewsService
+import de.tum.`in`.tumcampusapp.R
+import de.tum.`in`.tumcampusapp.component.tumui.calendar.CalendarController
+import de.tum.`in`.tumcampusapp.component.tumui.calendar.WidgetCalendarItem
+import de.tum.`in`.tumcampusapp.utils.Const
+import de.tum.`in`.tumcampusapp.utils.DateTimeUtils
+import org.joda.time.DateTime
+import org.joda.time.format.DateTimeFormat
+import java.util.*
+import kotlin.collections.ArrayList
+
+class TimetableWidgetService : RemoteViewsService() {
+
+    override fun onGetViewFactory(intent: Intent): RemoteViewsFactory {
+        return TimetableRemoteViewFactory(this.applicationContext, intent)
+    }
+
+    private inner class TimetableRemoteViewFactory internal constructor(private val applicationContext: Context, intent: Intent) : RemoteViewsFactory {
```

Comment:
This doesn’t need to be an inner class. Also, it’d might also be nice to add some line breaks 😉